virttest/vt_utils/block_disk.py

```diff
@@ -0,0 +1,408 @@
+#
+# library for block related helper functions
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; specifically version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat (c) 2023 and Avocado contributors
+# Author: Houqi Zuo <hzuo@redhat.com>
+import logging
+import platform
+import re
+
+from avocado.core import exceptions
```

Comment:
IMHO, the test-level exceptions shouldn't be raised by any of the functions in this module as they are utilities rather than test steps.